src/tribler-common/tribler_common/notifier.py

```diff
@@ -0,0 +1,167 @@
+import logging
+from asyncio import AbstractEventLoop
+from collections import defaultdict
+from contextlib import contextmanager
+from inspect import iscoroutinefunction, signature
+from threading import Lock
+from typing import Callable, Dict, TypeVar, cast
+
+
+FuncT = TypeVar("FuncT", bound=Callable[..., None])
+
+
+class NotifierError(Exception):
+    pass
+
+
+class Notifier:
+    def __init__(self, loop: AbstractEventLoop = None):
```

Comment:
**Question:** Are there situations in the Tribler core where we are sending messages across different `EventLoop`s?

Reply:
Yes, we use this "feature" inside Gigachannel Component.

An example: https://github.com/Tribler/tribler/blob/eed41c7d132f5f6709c8d89d8b25ae5b26700cdb/src/tribler-core/tribler_core/components/gigachannel_manager/gigachannel_manager.py#L294